src/components/Gallery/index.js

```diff
@@ -0,0 +1,188 @@
+
+  const updateImagesAndCursor = (result) => {
+    const newImages = result.edges.map(obj => obj.node.image.uri).filter((image) => !images.includes(image))
+    setImages([...images, ...newImages])
+    setHasMoreImages(result.page_info.has_next_page)
+    setEndCursor(result.page_info.end_cursor)
+  }
+
+  const onSubmitPress = useCallback(
+    () => {
+      const length = selectedImages.length
+      if (length >= minSelectionCount && length <= maxSelectionCount) {
+        onSubmit(selectedImages)
+      }
+    },
+    [selectedImages, params]
+  )
+
+  const loadMore = useCallback(
+    async () => {
+      const granted = await requestReadPermissions()
+      if (granted !== PermissionsAndroid.RESULTS.GRANTED) {
+        alertPermissionsError()
+        return
+      }
+
+      if (hasMoreImages) {
+        try {
+          updateImagesAndCursor(await getPhotos(endCursor))
+        } catch (error) {
+          alertPermissionsError()
+        }
+      }
+    },
+    [endCursor, hasMoreImages]
+  )
+
+  const canSelectMoreImages = useCallback(
+    () => selectedImages.length < maxSelectionCount,
+    [selectedImages]
+  )
+
+  const toggleSelect = useCallback(
+    (item) => {
+      const newSelectedImages = [...selectedImages]
+      const index = newSelectedImages.indexOf(item)
+
+      if (index !== -1) {
+        newSelectedImages.splice(index, 1)
+      } else {
+        if (canSelectMoreImages()) {
+          newSelectedImages.push(item)
+        }
+      }
+
+      setSelectedImages(newSelectedImages)
+    },
+    [selectedImages]
+  )
+
+  const callCamera = useCallback(
+    async () => {
+      const granted = await requestWritePermissions()
+
+      if (granted !== PermissionsAndroid.RESULTS.GRANTED) {
+        alertPermissionsError()
+
+        return
+      }
+
+      try {
+        const [image] = await ImageService.openCamera()
+        const newImages = [...images]
+
+        newImages.splice(1, 0, image)
+        await CameraRoll.saveToCameraRoll(image)
+        setImages(newImages)
+      } catch (error) {
+      }
+    },
+    [images]
+  )
+
+  useEffect(() => {
+    if (!params) return
+
+    if (params.addedImages) {
+      const newImages = [...images]
+      newImages.splice(1, 0, ...params.addedImages)
+      setImages(newImages)
+    }
+
+    if (params.selectedImages) {
+      setSelectedImages([...params.selectedImages])
+    }
+  }, [params])
+
+  const size = useMemo(
+    () => SCREEN_WIDTH / numColumns,
+    [numColumns]
+  )
+
+  const renderItem = useCallback(({ item, index }) => {
+    if (index === 0) {
+      return (
+        <TakePhotoButton onPress={callCamera} size={size} />
+      )
+    }
+
+    const toggleItem = () => toggleSelect(item)
+
+    return (
+      <ListItem
+        item={item}
+        width={size}
+        height={size}
+        selectedIndex={selectedImages.indexOf(item) + 1}
+        onPress={toggleItem}
+      />
+    )
+  })
```

Comment:
If you don't specify a dependencies Array, this will return a new callback on every render. The dependencies Array needs to be: `[size, toggleSelect, selectedImages]`